src/exception/users/notfound.error.ts

```diff
@@ -0,0 +1,13 @@
+import { HttpStatusCode } from '@constants/httpStatusCode';
+import { ErrorsMessages } from '@constants/errorMessages';
+import { BaseError } from '../base.error';
+
+export class UserNotFoundError extends BaseError {
```

Comment:
One tip here. It's a good practice to use the same file name as your class name, and also replace the spaces with a "-" for example this case could be "user-not-found.error.ts" instead of "notfound.error.ts".

This, besides being a good practice to keep a synergy between what you declare and what you name, also is going to help you when you need to search for a specific exception using your IDE. Probably, 99% of the IDEs will look for the file name because it's quite difficult for them to distinguish between class/interfaces/etc so if you wanna search "user exception" is not going to bring anything.